packages/dds/map/src/directory.ts

```diff
@@ -1380,12 +1448,13 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
      * @internal
      */
     public processDeleteSubDirectoryMessage(
+        msg: ISequencedDocumentMessage,
         op: IDirectoryDeleteSubDirectoryOperation,
         local: boolean,
         localOpMetadata: unknown,
     ): void {
         this.throwIfDisposed();
-        if (!this.needProcessSubDirectoryOperation(op, local, localOpMetadata)) {
+        if (!this.needProcessSubDirectoryOperation(msg, op, local, localOpMetadata)) {
             return;
         }
         this.deleteSubDirectoryCore(op.subdirName, local);
```

Comment:
deleteSubDirectoryCore will delete the subdirectory object entirely (and we'll only early-exit above if we have a pending subdirectory operation).  But suppose a third client C is watching two other clients A and B do subdirectory operations -- does client C need to keep the subdirectory object around for some time after client A deletes it to correctly interpret subsequent subdirectory operations from B?

Reply:
If I understand your question correctly, client C will just keep applying ops in order they are received. So, it will apply any keys ops received before the delete op from client A.